### PR TITLE
Make use of the extracted ErrorResponse

### PR DESCRIPTION
**Pull request by wallner**
The `ErrorResponse` class moved into the `scim-schema` preoject. This
commit makes use of the class from the `scim-schema`. It need osiam/scim-schema#152 to be accepted.

### CODE REVIEW

File: src/main/java/org/osiam/resources/exception/OsiamExceptionHandler.java [hunk 1]
```diff
@@ -101,45 +101,21 @@ public JsonErrorResult handleUnrecognizedProperty(Exception e) {
...
     }
 
-    private JsonErrorResult produceErrorResponse(String message, HttpStatus status) {
+    private ErrorResponse produceErrorResponse(String message, HttpStatus status) {
         return produceErrorResponse(message, status, null);
     }
 
-    private JsonErrorResult produceErrorResponse(String message, HttpStatus status,
+    private ErrorResponse produceErrorResponse(String message, HttpStatus status,
                                                  ErrorMessageTransformer transformer) {
```
Comment (tkrille) [inline src/main/java/org/osiam/resources/exception/OsiamExceptionHandler.java]: wrong indentation, either move it 2 spaces to the left or indent the line with 8 spaces.

**Comment by dacrome** [comment]
I triggered the tests again, the tests are green now.

**Comment by dacrome** [comment]
I guess you need to update also [the ITs](https://github.com/osiam/connector4java-integration-tests/blob/master/src/test/java/org/osiam/client/ErrorMessagesIT.java), right?

**Comment by tkrille** [comment]
LGTM, thanks.

**Pull request merged**